front-end/src/features/todo/todoSlice.ts

```diff
@@ -0,0 +1,47 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+
+import { TodoType } from "../../types";
+
+// Define a type for the slice state
+
+const initialState: TodoType[] = [];
+
+export const todoSlice = createSlice({
+    name: "todo",
+    initialState,
+    reducers: {
+        addTodoItemAction: (state, action: PayloadAction<TodoType>) => {
+            state.push(action.payload);
+        },
+        deleteTodoItemAction: (state, action: PayloadAction<TodoType>) => {
+            const index = state.findIndex(
+                (todo) => todo.id === action.payload.id
+            );
+            if (index !== -1) {
+                state.splice(index, 1);
+            }
+        },
+        setTodoItemAction: (state, action: PayloadAction<TodoType>) => {
+            const index = state.findIndex(
+                (todo) => todo.id === action.payload.id
+            );
+            if (index !== -1) {
+                state[index] = action.payload;
+            }
+        },
+        setTodoListAction: (state, action: PayloadAction<TodoType[]>) => {
```

Comment:
tried that exact logic already, for some unknown reason that doesn't work but this works

Reply:
In that case `return action.payload`.
Note: Slice actions have two options - 1. mutate the `state` and it will compare the changes itself, 2. return the `newState` and it will set that as the state value. Hence, using the second option.

Reply:
Ref: https://redux-toolkit.js.org/usage/immer-reducers#resetting-and-replacing-state